components/builder-protocol/protocols/vault.proto

```diff
@@ -56,101 +56,110 @@ message CheckOriginAccessRequest {
 
 message AccountInvitationListRequest {
-  required uint64 account_id = 1;
+  optional uint64 account_id = 1;
 }
 
 message AccountInvitationListResponse {
-  required uint64 account_id = 1;
+  optional uint64 account_id = 1;
   repeated OriginInvitation invitations = 2;
 }
 
 message OriginInvitationListRequest {
-  required uint64 origin_id = 1;
+  optional uint64 origin_id = 1;
 }
 
 message OriginInvitationListResponse {
-  required uint64 origin_id = 1;
+  optional uint64 origin_id = 1;
   repeated OriginInvitation invitations = 2;
 }
 
 message OriginInvitation {
-  required uint64 id = 1;
-  required uint64 account_id = 2;
-  required string account_name = 3;
-  required uint64 origin_id = 4;
-  required string origin_name = 5;
-  required uint64 owner_id = 6;
+  optional uint64 id = 1;
+  optional uint64 account_id = 2;
+  optional string account_name = 3;
+  optional uint64 origin_id = 4;
+  optional string origin_name = 5;
+  optional uint64 owner_id = 6;
 }
 
 message OriginInvitationCreate {
-  required uint64 account_id = 1;
-  required string account_name = 2;
-  required uint64 origin_id = 3;
-  required string origin_name = 4;
-  required uint64 owner_id = 5;
+  optional uint64 account_id = 1;
+  optional string account_name = 2;
+  optional uint64 origin_id = 3;
+  optional string origin_name = 4;
+  optional uint64 owner_id = 5;
 }
 
 message OriginInvitationAcceptRequest {
-  required uint64 account_accepting_request = 1;
-  required uint64 invite_id = 2;
-  required bool ignore = 3;
+  optional uint64 account_accepting_request = 1;
+  optional uint64 invite_id = 2;
+  optional bool ignore = 3;
+}
+
+message OriginInvitationValidateRequest {
+  optional uint64 account_accepting_request = 1;
+  optional uint64 invite_id = 2;
+}
+
+message OriginInvitationValidateResponse {
+  optional bool is_valid = 1;
 }
 
 message OriginInvitationAcceptResponse { }
 
 message OriginSecretKey {
-  required uint64 id = 1;
-  required uint64 origin_id = 2;
-  required string name = 3;
-  required string revision = 4;
-  required bytes body = 5;
-  required uint64 owner_id = 6;
+  optional uint64 id = 1;
+  optional uint64 origin_id = 2;
+  optional string name = 3;
+  optional string revision = 4;
+  optional bytes body = 5;
+  optional uint64 owner_id = 6;
 }
 
 message OriginSecretKeyCreate {
-  required uint64 origin_id = 1;
-  required string name = 2;
-  required string revision = 3;
-  required bytes body = 4;
-  required uint64 owner_id = 5;
+  optional uint64 origin_id = 1;
+  optional string name = 2;
+  optional string revision = 3;
+  optional bytes body = 4;
+  optional uint64 owner_id = 5;
 }
 
 message OriginSecretKeyGet {
-  required uint64 owner_id = 1;
-  required string origin = 2;
+  optional uint64 owner_id = 1;
+  optional string origin = 2;
 }
 
-message Project {
-  required string id = 1;
-  required uint64 owner_id = 2;
-  required string plan_path = 3;
-  oneof vcs {
-    VCSGit git = 4;
-  }
+message OriginProject {
```

Comment:
The `id` was `{name}/{origin}` before, this change means the messages will be slightly larger which is OK, but we need to assert that the name and origin don't differ from the combined field.

We also lose the typing around the VCS so we'll need to validate in the application code as well